### PR TITLE
Manually Disabling two TestDistBackendWithSpawn tests on ROCm, test_ddp_profiling_torch_profiler and test_ddp_sync_bn_training_vs_eval

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68255

Manually disabling these two tests because they can't be disabled via Probot.

See the issues #68222 and #68173 for details.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @jeffdaily @sunway513 @jithunnair-amd @ROCmSupport @KyleCZH

Differential Revision: [D32390899](https://our.internmc.facebook.com/intern/diff/D32390899)

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
Thank you! Do you also know if the test case _should_ inherit from common_utils.TestCase?

**Review by mrshenli — APPROVED** [review]

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32390899).

**Comment by mrshenli** [comment]
Hey @janeyx99,

The test is executed from `TestDistBackendWithSpawn`, which inherit from `common_utils.TestCase`. Is it possible to detect based on where the test is executed from instead of where the test implementation resides?

**Review by malfet — APPROVED** [review]

File: torch/testing/_internal/distributed/distributed_test.py [hunk 1]
```diff
@@ -5703,6 +5703,7 @@ def test_ddp_profiling_autograd_profiler(self):
             IS_MACOS or IS_WINDOWS,
             "torch.profiler not enabled for mac/windows: https://github.com/pytorch/pytorch/pull/56124",
         )
+        @skip_if_rocm
```
Comment (malfet) [inline torch/testing/_internal/distributed/distributed_test.py:5706]: Small nit: it would be good to mention tracking issue here

**Pull request closed without merging**